File: src/net/t3kt/tctrl/model/ModuleModel.java
package net.t3kt.tctrl.model;

import com.google.common.collect.ImmutableCollection;
import com.google.common.collect.ImmutableMap;
import javax.annotation.Nullable;
import net.t3kt.tctrl.schema.Groupable;
import net.t3kt.tctrl.schema.ModuleSchema;

public interface ModuleModel extends ParentModelNode<ModuleSchema>, Groupable {

    @Nullable
    ModuleModel getParentModule();

    ParamModel getParamByKey(String key);

    ImmutableCollection<? extends ParamModel> getParams();

    ImmutableCollection<? extends SingleParamModel> getFlatParams();

    ImmutableMap<String, ModelNodeGroup<ParamModel>> getParamGroups();
}

File: win/TctrlCore/Schema/Model/ModuleModel.cs
using System.Linq;

namespace Tctrl.Core.Schema.Model {
    public class ModuleModel : ParentModelNode<ModuleSchema> {
        public ModuleModel(ModuleSchema schema) : base(schema,
            schema.ChildModules.Select(module => new ModuleModel(module))) { }
    }
}